### PR TITLE
CB-16491 Refactor UserSyncService

### PR DESCRIPTION
**Pull request by lacikaaa**
- split into multiple classes
- cover everything with unit tests
- the logic is not changed

### CODE REVIEW

**Review by horadla23 — APPROVED** [review]
LGTM, someone from UserSyncService authors should check this too, @handavid can you check this?

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncForEnvService.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+    public void synchronizeUsers(String operationId, String accountId, List<Stack> stacks, UserSyncRequestFilter userSyncFilter,
+            UserSyncOptions options) {
+        operationService.tryWithOperationCleanup(operationId, accountId, () -> {
+            Set<String> environmentCrns = stacks.stream().map(Stack::getEnvironmentCrn).collect(Collectors.toSet());
+            UserSyncLogEvent logUserSyncEvent = options.isFullSync() ? FULL_USER_SYNC : PARTIAL_USER_SYNC;
+            LOGGER.info("Starting {} for environments {} with operationId {} ...", logUserSyncEvent, environmentCrns, operationId);
+
+            if (options.isFullSync()) {
+                umsVirtualGroupCreateService.createVirtualGroups(accountId, environmentCrns);
```
Comment (handavid) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncForEnvService.java:76]: Why is this needed? Aren't the virtual groups created/deleted as part of the environment lifecycle? 
The virtual groups, a.k.a. workload administration groups, will be returned by the UMS in either the DefaultUmsUsersStateProvider or the BulkUmsUsersStateProvider.

**Review by handavid — COMMENTED** [review]
Thank you for breaking this up!

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncForEnvService.java (hunk 1 above)
Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncForEnvService.java:76]: please see this commit: https://github.com/hortonworks/cloudbreak/commit/f9cab95e5346542b875e7e0572e6769e177a0e5b

Comment (horadla23) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncForEnvService.java:76]: we need it to support newly introduced virtual groups for existing environments

Comment (handavid) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncForEnvService.java:76]: ok. this conversation isn't really related to this PR. let's move the discussion to the other JIRA

**Review by handavid — APPROVED** [review]
